groups/nts/ntsu/ntsu_socketutil.cpp

```diff
@@ -3298,7 +3297,14 @@ ntsa::Error SocketUtil::receiveNotifications(
         msg.msg_control    = buf.buffer();
         msg.msg_controllen = k_BUF_SIZE;
 
-        ssize_t recvMsgResult = ::recvmsg(socket, &msg, MSG_ERRQUEUE);
+        const ssize_t recvMsgResult =
+            ::recvmsg(socket, &msg, MSG_ERRQUEUE | MSG_DONTWAIT);
+
+        // Check that if ::recvmsg returned 0 that there is really some data
+        // inside the control message
+        if (0 == recvMsgResult && 0 == CMSG_FIRSTHDR(&msg)) {
```

Comment:
Including `&& 0 == CMSG_FIRSTHDR(&msg)` seems suspicious. Simply breaking from the loop when `recvmsg` returns 0 seems sufficient. Have you reproduced a case where `recvmsg` returns 0 but the control data contains defined control messages?